### PR TITLE
Require stylelint v14 for stylelint config and scripts

### PR DESCRIPTION
**Pull request by ocean90**
## Description
This is a follow-up for #36518. As mentioned by @WraithKenny in https://github.com/WordPress/gutenberg/issues/36514#issuecomment-999922757 the current dependencies are not compatible with v14. Therefore we should increase the minimum peer dependency of `stylelint` to `14.2.0` and thus update scripts to use v14 too.

While working on this I noticed that the current stylelint repo in this config was not using the SCSS preset. This seems to be the reason why suddenly a few errors came up since I had to switch to the SCSS preset otherwise stylelint is no longer able to parse SCSS files, see [official migration guide to v14](https://github.com/stylelint/stylelint/blob/14.0.0/docs/migration-guide/to-14.md) for details.

To update dependencies the I have followed the steps mentioned in https://github.com/WordPress/gutenberg/blob/HEAD/packages/README.md.

## How has this been tested?
Manually by running `npm run lint-css` in the project.

## Types of changes
Breaking change

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/base-styles/_mixins.scss
Comment (ocean90) [inline packages/base-styles/_mixins.scss:222]:
```
packages/base-styles/_mixins.scss
 217:3  ✖  Expected single space after "}" of @if statement    scss/at-if-closing-brace-space-after
 224:3  ✖  Unexpected newline after "}" of @else statement     scss/at-else-closing-brace-newline-after
 224:3  ✖  Expected single space after "}" of @else statement  scss/at-else-closing-brace-space-after
````

File: packages/block-editor/src/components/link-control/style.scss
Comment (ocean90) [inline packages/block-editor/src/components/link-control/style.scss:64]:
```
packages/block-editor/src/components/link-control/style.scss
 64:37  ✖  Expected single space after "*" operator  function-calc-no-unspaced-operator
```

File: packages/scripts/package.json
Comment (ocean90) [inline packages/scripts/package.json:75]: Updated because `stylelint-config-recommended-scss` depends on `postcss-scss` which requires `postcss@^8.3.3`.

File: packages/stylelint-config/test/scss-valid.scss
Comment (ntwb) [inline packages/stylelint-config/test/scss-valid.scss:28]: With the addition of `packages/stylelint-config/test` in `.stylelintignore` in his PR at https://github.com/WordPress/gutenberg/pull/38091/files#diff-36ae73c99ebd1b4072a3164e014df72eb094703df4548dd51938aa7faaa7f8d2R1 are the ignore directives needed here?

**Review by ntwb — APPROVED** [review]
Thanks @ocean90, looks good to me, 1 comment, not a blocker though, approving and can update the PR if needed, otherwise merge away 💯

**Review by gziolo — APPROVED** [review]
Nice, thank you for bringing it up to date 🙇‍♂️

File: packages/stylelint-config/test/scss-valid.scss
Comment (ocean90) [reply packages/stylelint-config/test/scss-valid.scss:28]: Yes, the `.stylelintignore` is only used when running `npm run lint-css` in the project root but not when running `npm run test-unit packages/stylelint-config` for the tests.

The rule is new in v5, see https://github.com/stylelint-scss/stylelint-config-recommended-scss/commit/1e40cc1e1a288cc01aeeeccd8a0f72968bc9807a. Instead of removing the lines I added the comments for now.

**Comment by gziolo** [comment]
@ocean90, it looks like the recently merged Jest major version upgrade introduced merge conflict. The good news is that we should be up to date with most of the tools used in `@wordpress/scripts` once this PR lands 🎉

**Comment by ocean90** [comment]
I have rebased the branch. 🤞

File: packages/stylelint-config/test/scss-valid.scss
Comment (ntwb) [reply packages/stylelint-config/test/scss-valid.scss:28]: Cool, makes sense

**Pull request merged**
